test/neighborsearch.jl

```diff
@@ -0,0 +1,57 @@
+@testset "Neighborsearch" begin
+  @testset "KBallSearcher" begin
+    # KBallSearcher
+
+    #create a regular grid to test
+    pdomain = RegularGrid(10,10)
+
+    #test 1: all points should be found
+    maxneighbors = 10
+    neigh = BallNeighborhood(100.0)
+    searcher  = NeighborhoodSearcher(pdomain, neigh)
+    kballsearcher = KBallSearcher(searcher,maxneighbors)
```

Comment:
So the example I had in mind when you explained the KBallSearcher concept was something like:

```julia
knnsearcher = NearestNeighborSearcher(pdomain, maxneighbors)
kballsearcher = KBallSearcher(knnsearcher, neigh)
```
So the inner searcher is always a `NearestNeighborSearcher` and the other missing step is the `neigh` (ball). Makes sense?